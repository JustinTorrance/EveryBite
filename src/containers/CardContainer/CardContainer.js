import React, { Component } from 'react'
import { connect } from 'react-redux'
import RecipeCard from '../../components/RecipeCard/RecipeCard'
import shortid from 'shortid'

export class CardContainer extends Component {
  
  render() {
    
    let displayCards = this.props.recipes.map(recipe => {
      return <RecipeCard {...recipe} key={shortid.generate()} />
     })
    
    return(
      <div className='card-container'>
        {displayCards}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  recipes: state.recipes
})

export default connect(mapStateToProps)(CardContainer)