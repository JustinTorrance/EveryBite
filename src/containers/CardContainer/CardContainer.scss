.card-container {
  height: 100%;
  display: grid;
  grid-template-columns: 21% 21% 21%;
  grid-template-rows: auto;
  justify-content: center;
  column-gap: 5%;
  row-gap: 2.5%;
  margin-top: 50px;
}

.home-btn-div {
  display: flex;
  width: 100%;
  justify-content: center;
}

.no-recipes {
  font-family: 'Pacifico', cursive;
  // color: white;
  font-size: 2rem;  
}