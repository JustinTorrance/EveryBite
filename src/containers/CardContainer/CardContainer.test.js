import { CardContainer, mapStateToProps } from './CardContainer'

describe('CardContainer', () => {
  
  describe('mapStateToProps', () => {
    
    it('should return an object with isLoading and error', () => {
      const mockState = {
        isLoading: false,
        error: 'Something went wrong',
        recipes: [{name: 'curry'}]
      }
      const expected = {
        isLoading: false,
        recipes: [{name: 'curry'}]
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
})