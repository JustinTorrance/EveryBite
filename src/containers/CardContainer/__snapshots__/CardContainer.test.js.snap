// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
"<section className=\\"card-section\\">
  <div className=\\"home-btn-div\\">
    <Link to=\\"/\\">
      <WithStyles(Button) type=\\"submit\\" variant=\\"contained\\" color=\\"primary\\" className=\\"home-btn\\">
        Home
      </WithStyles(Button)>
    </Link>
  </div>
  <div className=\\"card-container\\">
    <RecipeCard name=\\"curry\\" />
  </div>
</section>"
`;
